extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/client/HTTPAppFactoryClient.java

```diff
@@ -0,0 +1,224 @@
+
+        //Creating a http post request by giving server URL
+        HttpPost httpLoginRequest = new HttpPost(serverURL);
+        HttpResponse httpLoginResponse;
+
+        try {
+            httpLoginRequest.setEntity(createEntityFromRequestParams(loginRequestParams));
+        } catch (UnsupportedEncodingException e) {
+            log.error("Error in encoding login request parameters to an entity, Parameters : " + loginRequestParams, e);
+            throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+        }
+
+        //Sending HTTP post request to AppFactory/ AppCloud
+        try {
+            httpLoginResponse = client.execute(httpLoginRequest);
+        } catch (IOException e) {
+            log.error("Error while sending HTTP Post request to server, Server URL : " + serverURL
+                    + ", HTTP Request : " + httpLoginRequest, e);
+            throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+        }
+
+        if (200 == httpLoginResponse.getStatusLine().getStatusCode()) {    //If response status is 200(OK)
+            //Extracting cookie value from response
+            cookie = httpLoginResponse.getFirstHeader("Set-Cookie").getValue().split(";")[0];
+
+            HttpEntity loginResponseEntity = httpLoginResponse.getEntity();
+            BufferedReader bufferedReader;
+            StringBuilder responseBuilder = new StringBuilder();
+            String responseLine;
+
+            try {
+                bufferedReader = new BufferedReader(new InputStreamReader(loginResponseEntity.getContent()));
+            } catch (IOException e) {
+                log.error("Unable to create an input stream from HTTP response received", e);
+                throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+            }
+
+            try {
+                while ((responseLine = bufferedReader.readLine()) != null) {
+                    responseBuilder.append(responseLine);
+                }
+            } catch (IOException e) {
+                log.error("Error in reading login response", e);
+                throw new AppFactoryServerException("Unable to login user, Details : " + loginRequestParams, e);
+            }
+            successLogin = responseBuilder.toString();
+
+            if ("true".equals(successLogin)) {
+                loginResponse.setLoggedIn(true);
+                loginResponse.setErroneousRequest(false);
+            } else {
+                loginResponse.setLoggedIn(false);
+                loginResponse.setErroneousRequest(true);
+                loginResponse.setErrorType(ErrorType.INVALID_CREDENTIALS);
+            }
+
+            try {
+                EntityUtils.consume(loginResponseEntity);
+            } catch (IOException e) {
+                log.error("Error in closing Entity stream of App Factory login response", e);
```

Comment:
Yes, I did this because causing an error when closing an entity stream won't affect any results from the method, so I just logged the error and continue (without terminating the method by throwing an exception to upstream).WDYT?